### ISSUE TITLE
Refactor the `AudienceSegmentationSetupCTAWidget` to not display simultaneously with other CTAs

### ISSUE THREAD

**Issue by jimmymadon** [google/site-kit-wp#9886]
## Feature Description

Since the introduction of multiple recent features and modules, their new setup CTAs can all be rendered simultaneously if their necessary feature flags and other conditions are met.

Until the introduction and overhaul of our existing notifications into perhaps a new "notifications centre", we should fix this issue to only show one CTA at a time. This issue focuses on refactoring the `AudienceSegmentationSetupCTAWidget` to use the Setup CTAs queue created as part of the `SignInWithGoogleSetupCTA` component.

https://github.com/user-attachments/assets/808cacca-f319-470a-b727-117870d3a748

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* The `AudienceSegmentationSetupCTAWidget` should be refactored to render in a separate queue of setup CTAs.
* This CTA should be rendered with the highest priority, over and above other Setup CTAs. 
* After all Setup CTAs are refactored, the order should be:
    * `AudienceSegmentationSetupCTAWidget`
    * `ConsentModeSetupCTAWidget`
    * `FirstPartyModeSetupBanner`
    * `SignInWithGoogleSetupCTABanner`
    * `ReaderRevenueManagerSetupCTABanner`

## Implementation Brief

Note: Follow the pattern in the PR for #9889 and wait for it to be merged.

* [ ] Register the `AudienceSegmentationSetupCTAWidget` as a separate notification within `assets/js/modules/analytics-4/index.js`: 
  * [ ] Use `audiences-setup-cta-dashboard` as the notification ID as this is the ID used in for the GA tracking events.
  * [ ] Use `groupID: NOTIFICATION_GROUPS.SETUP_CTAS`.
  * [ ] The priority number here should be 10. Since Setup CTAs have their own queue (determined by their own groupID), having these lower priorities will not affect other notifications with the same or lower priority.
  * [ ] For the `Component`, use the existing `AudienceSegmentationSetupCTAWidget` when registering the notification and remove the direct call from `<DashboardMainApp>`. 
  * [ ] The `checkRequirements` callback should include all logic from within the notification (`assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget/index.js`) which prevents it from rendering (i.e. checks that return `null`.)
  * [ ] Set `dismissRetries` to 1.
* [ ] In `assets/js/modules/analytics-4/components/audience-segmentation/dashboard/AudienceSegmentationSetupCTAWidget/index.js`: 
  * [ ] Remove logic for tracking GA events for the standard notification events.
  * [ ] Remove the usage of the `withIntersectionObserver` HOC and ensure the logic is covered by the `ViewedStateObserver`.
  * [ ] Refactor the component to use the new `NotificationWithSVG` component.
  * [ ] Use the `dismissNotification` and `isNotificationDismissed` actions / selectors instead of directly using `prompts`. 

### Test Coverage

* Existing tests should be refactored.

## QA Brief

* Setup Site Kit and Analytics 4 module
* Verify that Audiences setup CTA banner is showing properly
* Verify that interacting with CTAs is generating tracking
* Verify that clicking `Maybe later` CTA is opening the tooltip and `view_tooltip` is correctly tracked
* After dismissed with `Maybe later` go to `wp_usermeta` table and for current user ID edit `wp_googlesitekitpersistent_dismissed_prompts` column. Find the part (should be tha last line), that will look something like this `s:44:"audience_segmentation_setup_cta-notification";a:2:{s:7:"expires";i:1737999730;s:5:"count";i:1;}`. Edit the `i:1737999730` by replacing the timestamp to something in the past, like this one `1736699414` and verify that banner is showing again with `Don't show again` CTA
* Clicking `Don't show again` CTA should dismiss banner and `expires` timestamp should be showing zero value, like `s:7:"expires";i:0;` and not showing the banner on next page load

Verify that error modal variants are correctly showing:

### OAuth Error Variant

To simulate the OAuth error variant:

1. **Prepare the Environment:**
   - Delete the `wp_googlesitekit_additional_auth_scopes` entry from the `wp_usermeta` table if it exists. This entry contains the `https://www.googleapis.com/auth/analytics.edit` scope.
   
2. **Trigger the OAuth Error:**
   - Click on the **Enable groups** button in the Audience Segmentation setup CTA banner.
   - In the OAuth flow, when prompted to grant permissions, cancel the flow.
   - Ensure the OAuth error variant of the Error Modal is displayed.

### Insufficient Permissions Error Variant

To simulate the Insufficient Permissions error variant:

1. **Install and Configure Tweak Extension:**
   - Install the [Tweak](https://chromewebstore.google.com/detail/tweak-mock-and-modify-htt/feahianecghpnipmhphmfgmpdodhcapi?hl=en) Chrome extension.
   - Add a new rule to block the `sync-audiences` request:
     - URL: `.*/wp-json/google-site-kit/v1/modules/analytics-4/data/sync-audiences*`
     - Enable the use of regular expression `(.*)`
     - HTTP Method: `POST`
     - Status code: `403`
     - Response payload:
       ```json
       {
         "code": 403,
         "message": "Insufficient Permissions Test Error",
         "data": {
           "status": 403,
           "reason": "insufficientPermissions"
         }
       }
       ```

2. **Trigger the Insufficient Permissions Error:**
   - Click on the **Enable groups** button in the Audience Segmentation setup CTA banner.
   - Ensure the Insufficient Permissions error variant of the Error Modal is displayed.

### Generic Error Variant

   * **Simulate Sync Available Audiences API Error:**
     - Add a new rule in the Tweak extension to block the `sync-audiences` request:
       - URL: `.*/wp-json/google-site-kit/v1/modules/analytics-4/data/sync-audiences*`
       - Enable the use of regular expression `(.*)`
       - HTTP Method: `POST`
       - Status code: `500`
       - Response payload:
         ```json
         {
           "code": "internal_server_error",
           "message": "Internal server error",
           "data": { "status": 500 }
         }
         ```

   * **Trigger the Generic Error:**
     - Click on the **Enable groups** button in the Audience Segmentation setup CTA banner.
     - Ensure the Generic error variant of the Error Modal is displayed.

## Changelog entry

* Update the Audience Segmentation Setup CTA so it does not appear alongside other CTAs.

**Comment by eugene-manuilov** [comment]
AC ✔

**Comment by eugene-manuilov** [comment]
> * Register the `AudienceSegmentationSetupCTAWidget` as a separate notification:

@jimmymadon, let's make it clear where it should be registered, in which file.

**Comment by eugene-manuilov** [comment]
IB ✔

**Comment by kelvinballoo** [comment]
### QA Update ❌ 

I did another test and the spinner is now loading.
That said, I noticed that after clicking 'Enable groups', the subtle success notification doesn't appear. It's only after I reload the page that it appears.
This is not as expected.

Video is attached for reference:
<Details>

https://github.com/user-attachments/assets/99c76fc9-4c98-443b-8c41-978215f7bab4

</Details>

**Comment by zutigrm** [comment]
Thanks @kelvinballoo , this is separate notification from the setup CTA refactored here. Doesn't seem related to any work done as part of this issue. This seems to be outstanding/known issue, as there is a follow up to fix that behaviour #9453

**Comment by techanvil** [comment]
Hey @zutigrm, I see you've raised a followup PR for this issue.

Please could you amend it to also ensure the Audience Segmentation Setup CTA Notification is only actually registered if the feature flag is enabled? The previous PR introduced a regression whereby the audience settings are being requested even when the feature isn't enabled, as per this [Slack thread](https://fueled10up.slack.com/archives/CBKKQEBR9/p1738276898451329). The change that introduced is is the one discussed [here](https://github.com/google/site-kit-wp/pull/10017/files#r1925646967).

**Comment by kelvinballoo** [comment]
### QA Update ✅ 

The audience success notification now appears right after enabling the visitor groups. This is as expected. ✅ 
<Details>

https://github.com/user-attachments/assets/0561e868-c72a-4567-9907-88fd0396f71e

</Details>


Other than that the QAB was tested good.
<Details>
Tooltip tracking:

https://github.com/user-attachments/assets/7f7a3d90-5923-425a-8b03-0f2f10d2b04d

<img width="1204" alt="Image" src="https://github.com/user-attachments/assets/f7aebf0c-decf-4590-9f31-1b0e9b9f355a" />

Clicking 'Don't show again' and set expiry to 0:

https://github.com/user-attachments/assets/de2b8252-a650-456d-b392-185d494d2c00


AS Setup banner appears without any other setup CTA:

https://github.com/user-attachments/assets/7fb788de-9197-4826-8495-1d61c8c6572f


OAuth error variant:

https://github.com/user-attachments/assets/876a16a4-7f0a-4ac7-9b4f-856613dcefe1


Insufficient permission error:

https://github.com/user-attachments/assets/8fa094b0-28c9-4943-84f9-4aff1493f130


Generic error:

https://github.com/user-attachments/assets/989e1560-da07-4593-9efc-1d5c4f26c7eb

</Details>

**Comment by zutigrm** [comment]
Thanks @kelvinballoo 

> However, I do want to check if there is any specific test to be done based on Tom's comment https://github.com/google/site-kit-wp/issues/9886#issuecomment-2626884789 ?

The change Tom outlined is to prevent AS requests when audienceSegmentation feature flag is disabled - meaning no console error is shown 

> The way I did a test is to set up SK and enabled a couple of feature flags except Audience Segmentation.

> I was expecting the AS banner to show up first but FPM showed up.

Since you did not enable `audienceSegmentation` feature flag, AS banner will not show, that is expected behaviour, it will only show when feature flag is enabled

**Comment by kelvinballoo** [comment]
### QA Update ✅ 

Thanks @zutigrm 

Verified that there were no console errors on the Dashboard on develop and this is as expected.

<Details>

https://github.com/user-attachments/assets/41280109-9794-41a6-ab79-af018849a454

</Details>

___________________________________________________________________________

Other than that the QAB was tested good.
<Details>
Tooltip tracking:

https://github.com/user-attachments/assets/7f7a3d90-5923-425a-8b03-0f2f10d2b04d

<img width="1204" alt="Image" src="https://github.com/user-attachments/assets/f7aebf0c-decf-4590-9f31-1b0e9b9f355a" />

Clicking 'Don't show again' and set expiry to 0:

https://github.com/user-attachments/assets/de2b8252-a650-456d-b392-185d494d2c00


AS Setup banner appears without any other setup CTA:

https://github.com/user-attachments/assets/7fb788de-9197-4826-8495-1d61c8c6572f


OAuth error variant:

https://github.com/user-attachments/assets/876a16a4-7f0a-4ac7-9b4f-856613dcefe1


Insufficient permission error:

https://github.com/user-attachments/assets/8fa094b0-28c9-4943-84f9-4aff1493f130


Generic error:

https://github.com/user-attachments/assets/989e1560-da07-4593-9efc-1d5c4f26c7eb

</Details>
